tests/CMakeLists.txt

```diff
@@ -1,132 +1,142 @@
-﻿#=============================================================================
+# =============================================================================
 # Copyright (c) 2018-2020, NVIDIA CORPORATION.
 #
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+# http://www.apache.org/licenses/LICENSE-2.0
 #
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#=============================================================================
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
-
-project(RMM_TESTS LANGUAGES C CXX CUDA)
-
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-
-set(CMAKE_CUDA_STANDARD 14)
-set(CMAKE_CUDA_STANDARD_REQUIRED ON)
-
-###################################################################################################
-# - compiler function
-
-function(ConfigureTest CMAKE_TEST_NAME CMAKE_TEST_SRC)
-  add_executable(${CMAKE_TEST_NAME} ${CMAKE_TEST_SRC})
-  set_target_properties(${CMAKE_TEST_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON)
-  target_include_directories(${CMAKE_TEST_NAME} PRIVATE "$<BUILD_INTERFACE:${RMM_SOURCE_DIR}>")
-
-  target_link_libraries(${CMAKE_TEST_NAME} GTest::gmock GTest::gtest GTest::gmock_main GTest::gtest_main
-                        pthread rmm)
-  
-  set_target_properties(${CMAKE_TEST_NAME} PROPERTIES
-                        RUNTIME_OUTPUT_DIRECTORY "${RMM_BINARY_DIR}/gtests")
-
-  target_compile_definitions(${CMAKE_TEST_NAME} PUBLIC
-                        "SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${RMM_LOGGING_LEVEL}")    
-
-  add_test(NAME ${CMAKE_TEST_NAME} COMMAND ${CMAKE_TEST_NAME})
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+
+include(CTest)
+
+# Fetch GTest
+
+CPMFindPackage(
+  NAME GTest
+  GITHUB_REPOSITORY google/googletest
+  GIT_TAG release-1.10.0
+  VERSION 1.10.0
+  GIT_SHALLOW TRUE
+  OPTIONS "INSTALL_GTEST OFF"
+  # googletest >= 1.10.0 provides a cmake config file -- use it if it exists
+  FIND_PACKAGE_ARGUMENTS "CONFIG")
+
```

Comment:
I'm not sure that it's such a great idea to scatter the thirdparty stuff around, instead of having it all in one place. In particular, I seem to remember that googlebenchmark requires googletest, but I could be wrong on that (and I think if so, it's already not correctly handled right now).

Reply:
We're only doing it for GTest / Google Benchmark.

Google Benchmark does not depend on GTest so we should be safe to do this.

Reply:
Right, it doesn't because we have `BENCHMARK_ENABLE_TESTING OFF`. I had hit a case lately where I was building benchmark separately, and at that point it complained about missing gtest, but it's not a problem for the usage here.